src/js/portal/content/itemData.js

```diff
@@ -1,6 +1,17 @@
 import request from "../request";
 
-export function itemData(id) {
+export function itemData(id, name) {
+    // If name is passed as an optional parameter, check to see if it
+    //  contains a period.  This is to handle instances where file based
+    //  items aren't displaying JSON properly, trying to download instead.
+    if ((name !== null) && (name !== undefined)) // should handle null and undefined
+    {
```

Comment:
Since the goal here is to prevent the app from making the call to `/data` I think it would make more sense to put the logic in `main.js` and not in `portal.js`. The latter is meant to be a fairly pure wrapper around the REST API.